src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IArgument.cs

```diff
@@ -1381,8 +1381,7 @@ void M2(int x = ""string"")
     IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: P, IsImplicit) (Syntax: 'M2')
   Arguments(1):
       IArgumentOperation (ArgumentKind.DefaultValue, Matching Parameter: x) (OperationKind.Argument, Type: null, IsImplicit) (Syntax: 'M2()')
-        IInvalidOperation (OperationKind.Invalid, Type: System.Int32, IsImplicit) (Syntax: 'M2()')
-          Children(0)
+        IDefaultValueOperation (OperationKind.DefaultValue, Type: System.Int32, Constant: 0, IsImplicit) (Syntax: 'M2()')
```

Comment:
Sounds good. I think I got myself mixed up when reviewing the error behaviors in IOperation and emit respectively.

I was working under the assumption that the original behavior for emit was that bad constant values from metadata would be ignored and silently replaced with `default(ParameterType)`. IOperation was different because it actually exposed the bad constant value for the default argument. As part of unifying the process of generating the default arguments it felt right to expose the same information the compiler would use in order to emit.

However after reviewing the existing tests more carefully I realized that we actually *do* give an error when a constant value fails to convert to the parameter type, as you can see in `InvalidConversionForDefaultArgument_InIL`, **except** when the constant is a decimal or datetime. See `CodeGenConversionTests.DefaultParameterValue_DateTimeConstant` and `CodeGenConversionTests.DefaultParameterValue_DecimalConstant`.

It feels like in cases where the constant value is "known", i.e. not `ConstantValue.Bad`, and it fails to convert to the parameter type, and it's not the silly special decimal or datetime cases, then we should just give a BadExpression for the argument in binding and thereby satisfy these baselines.

Reply:
After some discussion offline we concluded that this particular scenario actually *should* include a default value in the operation tree. This behavior is happening because the parameter with the bad default value is from source. I think this might be done to reduce cascading diagnostics, but I'm not certain. I do know that tests like [BadDefaultArgumentInOtherAssembly](https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Compilers/CSharp/Test/Emit/Emit/EmitErrorTests.cs#L89) expect this behavior, and that as part of this change we're unifying IOperation with the compiler's emit phase to show what the compiler is really putting there for the argument.

I'm adding some comments to the test to try and explain and demonstrate the differences when calling methods with bad parameter default values from source and IL.